pkg/cmd/server/start.go

```diff
@@ -63,15 +66,17 @@ func (o CustomMetricsAdapterServerOptions) Config() (*apiserver.Config, error) {
 		return nil, err
 	}
 
-	if err := o.Authentication.ApplyTo(&serverConfig.Authentication, serverConfig.SecureServing, serverConfig.OpenAPIConfig); err != nil {
+	if err := o.Authentication.ApplyTo(&serverConfig.Authentication, serverConfig.SecureServing, nil); err != nil {
 		return nil, err
 	}
 	if err := o.Authorization.ApplyTo(&serverConfig.Authorization); err != nil {
 		return nil, err
 	}
 
-	// TODO: we can't currently serve swagger because we don't have a good way to dynamically update it
-	// serverConfig.SwaggerConfig = genericapiserver.DefaultSwaggerConfig()
+	// enable OpenAPI schemas
+	serverConfig.OpenAPIConfig = genericapiserver.DefaultOpenAPIConfig(generatedopenapi.GetOpenAPIDefinitions, openapinamer.NewDefinitionNamer(apiserver.Scheme))
+	serverConfig.OpenAPIConfig.Info.Title = "custom-metrics-adapter"
+	serverConfig.OpenAPIConfig.Info.Version = "1.0.0" // TODO(directxman12): remove this once autosetting this doesn't require security definitions
```

Comment:
I guess the above three should be made somehow customisable by users of this library, what would be the best approach for this?

Reply:
Maybe I am missing something. It is my understanding that the `custom` part of this project is with regard to where the metrics come from. Projects implementing this and providing such metrics would still utilize the `MetricValueList/custom.metrics.k8s.io` and `ExternalMetricValueList/external.metrics.k8s.io` resource types. In that sense, this library could still provide those types.